### PR TITLE
Upgrade to Apache HttpClient 5.x

### PR DESCRIPTION
**Pull request by otrosien**
Upgrades from Apache HttpClient 4.x to HttpClient 5.x (classic API).

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -31,4 +31,4 @@ jobs:
       - name: Test oldest supported library versions
         run: gradle build -Pjackson.version=2.8.0 -Pspring.version=6.0.0 -Pspringboot.version=3.0.0 -Pokhttp.version=3.3.0 -Papachehttp.version=4.4
```
Comment (otrosien) [inline .github/workflows/ci.yml:32]:
```suggestion
        run: gradle build -Pjackson.version=2.8.0 -Pspring.version=6.0.0 -Pspringboot.version=3.0.0 -Pokhttp.version=3.3.0 -Papachehttp.version=5.0
```

File: fahrschein-http-apache/src/main/java/org/zalando/fahrschein/http/apache/HttpComponentsRequest.java [hunk 2]
```diff
@@ -55,7 +55,11 @@ public String getMethod() {
 
     @Override
     public URI getURI() {
-        return this.httpRequest.getURI();
+        try {
+            return this.httpRequest.getUri();
+        } catch (URISyntaxException e) {
```
Comment (otrosien) [inline fahrschein-http-apache/src/main/java/org/zalando/fahrschein/http/apache/HttpComponentsRequest.java:60]: This one is mean – throwing a syntax exception on the getter, and not on the setter. 😓

Comment (MALPI) [reply fahrschein-http-apache/src/main/java/org/zalando/fahrschein/http/apache/HttpComponentsRequest.java:60]: Should we file a Technical Debt ticket for it and handle it in a separate PR?

File: fahrschein-e2e-test/src/test/java/org/zalando/fahrschein/e2e/NakadiTestWithDockerCompose.java [hunk 3]
```diff
@@ -92,7 +92,14 @@ private String postJson(HttpEntity e, URI uri) throws IOException {
         HttpPost post = new HttpPost(uri);
         post.setEntity(e);
         post.setHeader("Content-Type", "application/json");
-        return httpClient.execute(post, new BasicResponseHandler());
+        return httpClient.execute(post, response -> {
+            if (response.getCode() >= 300) {
```
[the remark this answers is not in the document]
Comment (otrosien) [reply fahrschein-e2e-test/src/test/java/org/zalando/fahrschein/e2e/NakadiTestWithDockerCompose.java:96]: I copied this one-to-one from the old Apache source. It's anyway just for testing, so failing for an unexpected redirect is also good IMO.

File: fahrschein-http-apache/src/main/java/org/zalando/fahrschein/http/apache/HttpComponentsRequest.java (hunk 2 above)
Comment (otrosien) [reply fahrschein-http-apache/src/main/java/org/zalando/fahrschein/http/apache/HttpComponentsRequest.java:60]: No, I don't think there's anything we can do here. The URISyntaxException is coming from Apache, and I really don't want to expose it like this in our own API definition.

File: fahrschein-http-apache/src/main/java/org/zalando/fahrschein/http/apache/HttpComponentsRequest.java [hunk 4]
```diff
@@ -55,7 +55,11 @@ public String getMethod() {
...
+            throw new RuntimeException(e);
```
[the remark this answers is not in the document]
Comment (otrosien) [reply fahrschein-http-apache/src/main/java/org/zalando/fahrschein/http/apache/HttpComponentsRequest.java:61]: Any other idea?

File: fahrschein-http-apache/src/main/java/org/zalando/fahrschein/http/apache/HttpComponentsRequest.java (hunk 2 above)
Comment (otrosien) [reply fahrschein-http-apache/src/main/java/org/zalando/fahrschein/http/apache/HttpComponentsRequest.java:60]: .. we could eliminate the problem via `URI.create(this.httpRequest.getRequestUri());`

**Comment by MALPI** [comment]
👍

**Comment by otrosien** [comment]
👍

**Pull request merged**